### PR TITLE
DMP-4055 Retention Attributes Not Applied to Media, Annotations, Transcriptions, and Case Document

### PR DESCRIPTION
**Pull request by karen-hedges**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/DMP-4055

### Change description ###

Updated the case document generation to also reset retention data

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x ] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/darts/casedocument/service/impl/GenerateCaseDocumentSingleCaseProcessorImplTest.java [hunk 1]
```diff
@@ -137,6 +143,8 @@ void testGenerationOfCaseDocument() {
         assertThat(savedCaseDocument.getCreatedBy()).isEqualTo(user);
         assertThat(savedCaseDocument.getLastModifiedBy()).isEqualTo(user);
         assertThat(savedCaseDocument.isHidden()).isEqualTo(false);
+
+        verify(caseRepository).save(courtCaseCaptor.capture());
```
Comment (Ben-Edwards-cgi) [inline src/test/java/uk/gov/hmcts/darts/casedocument/service/impl/GenerateCaseDocumentSingleCaseProcessorImplTest.java:147]: Should validate content of the capture to make sure the correct fields are being set on the object

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/GenerateCaseDocumentProcessorIntTest.java [hunk 2]
```diff
@@ -78,6 +80,11 @@ void testGenerateCaseDocument() {
 
         List<ExternalObjectDirectoryEntity> eodCaseDocument = eodRepository.findByCaseDocument(caseDocument);
         assertThat(eodCaseDocument.size()).isEqualTo(1);
+
+        caseRepository.findById(courtCase.getId()).ifPresent(c -> {
+            assertThat(c.isRetentionUpdated()).isTrue();
+            assertThat(c.getRetentionRetries()).isEqualTo(0);
+        });
```
Comment (Ben-Edwards-cgi) [inline src/integrationTest/java/uk/gov/hmcts/darts/task/service/GenerateCaseDocumentProcessorIntTest.java:87]: Should add a generic catch incase this id could not be found fail the test

**Review by Ben-Edwards-cgi — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/darts/casedocument/service/impl/GenerateCaseDocumentSingleCaseProcessorImplTest.java (hunk 1 above)
Comment (karen-hedges) [reply src/test/java/uk/gov/hmcts/darts/casedocument/service/impl/GenerateCaseDocumentSingleCaseProcessorImplTest.java:147]: tried but can't as the save method return is mocked so you are basically setting the values and comparing the same which makes it pointless.

File: src/integrationTest/java/uk/gov/hmcts/darts/task/service/GenerateCaseDocumentProcessorIntTest.java (hunk 2 above)
Comment (karen-hedges) [reply src/integrationTest/java/uk/gov/hmcts/darts/task/service/GenerateCaseDocumentProcessorIntTest.java:87]: technically you have a non null on case so if case document can be found so can case but I can add it just to bomb proof it

**Review by hemantasharma1129 — APPROVED** [review]

**Review by Ben-Edwards-cgi — APPROVED** [review]

**Pull request merged**